Add New Ballerina Sample with Component yaml (#193)
wso2/choreo-samples: patient-management-service/README.md

# Patient Management Service (Mediflow) - Ballerina

> Sample for Patient Management Service service.

Reviewer: Would be good to give a meaningful intro here. Similar to https://github.com/wso2/choreo-samples/tree/e641ad1480f18bb11f5fbb7039cb1ba12299e105/employee-details-service#introduction

### Prerequisites
1. Fork the repositoy

> ## Getting started

Reviewer: Shall we go with **Deploy in Choreo**? Suggesting so that the titles are consistent across other similar samples.

Please refer to the Choreo documentation under the [Develop an Application with Buildpacks](https://wso2.com/choreo/develop-components/deploy-an-application-with-buildpacks) to learn how to deploy the application.

1. Click `Create` button in the Project Overview Page.
2. Select `Service` Card from Component Creation Wizard.
3. Select `Ballerina` as the buildpack. Fill as follow according to selected Buildpack.

    | **Field**             | **Description**                               |
    |-----------------------|-----------------------------------------------|
    |Name           | Patient Management Ballerina Service |
    |Description    | Patient Management (Mediflow) Ballerina Service  |
    | **GitHub Repository** | choreo-samples (Click **`Try with Sample URL`**) |
    | **Branch**            | **`main`**                               |
    | **Buildpack**      | Ballerina|
    | **Select Ballerina Project Directory**       | patient-management-service |

4. Click Create. Once the component creation is complete, you will see the component overview page.
5. Deploy the created component.

> ## Testing

Reviewer: Suggestion: 
Test the service

The **MediFlow** service provides functionalities to manage patient data, including:
- Adding a new patient
- Retrieving patient details by name
- Listing all patients

You can test the service in the `Test Console` Page. You can find it in the side drawer.

### 1. Health Check
Verify if the service is operational:

**Endpoint:** `/health`
**Method:** `GET`

📌 Response:
```bash
{
  "status": "MediFlow is operational"
}
```

> ### 2. Add a New Patient

Reviewer: ```suggestion
### 2. Add a new patient
```

**Endpoint:** `/patients`  
**Method:** `POST` 

```bash
{
  "name": "Alice",
  "age": 30,
  "condition": "Healthy"
}
```
📌 Response:
```bash
{
  "message": "Patient added",
  "patient": "{
    "name": "Alice",
    "age": 30,
    "condition": "Healthy"
  }"
}
```

> ### 3. Retrieve a Patient by Name

Reviewer: ```suggestion
### 3. Retrieve a patient by name
```

**Endpoint:** `/patients/{name}`  
**Method:** `GET`  

📌 Response:
```bash
{
  "name": "Alice",
  "age": 30,
  "condition": "Healthy"
}
```

> ### 4. List All Patients

Reviewer: ```suggestion
### 4. List all patients
```

# Patient Management Service (Mediflow) - Ballerina

> This is a simple patient management REST service developed using Ballerina. This service provides functionalities to manage patient data, including: adding a new patient, retrieving patient details by name, and listing all patients.

Reviewer: ```suggestion
This is a simple patient management REST service developed using Ballerina. This service provides functionalities to manage patient data, including adding a new patient, retrieving patient details by name, and listing all patients.
```

## Deploy in Choreo

Follow the following steps to deploy the application in Choreo.

1. Create a component

>      - Fork this repository

Reviewer 2: Let's remove this steps as it is not mandatory

> 

Reviewer 2: It looks like this link is not working. Please check.

Also, what if we mention this at the end of this section as additional information? I believe the user might get confused about whether to follow the instructions here or in the Choreo docs.

Author: Updated.

     - Login to [Choreo](https://wso2.com/choreo/)

>      - Navigate to create a `Service` component

Reviewer: ```suggestion
     - If you already have one or more components in your project, click **+ Create**. Otherwise, click the **Service** card. 
```

>      - Provide a name and description for the component

Reviewer: ```suggestion
     - Provide a name and description for the component.
```

>      - Click **`Try with Sample URL`**

Reviewer: ```suggestion
     - Click **`Try with Sample URL`**.
```

>      - Select `Ballerina` as `Buildpack`

Reviewer: ```suggestion
     - Select `Ballerina` as the `Buildpack`.
```

>      - Select `patient-management-service` as the `Ballerina Project Directory`

Reviewer: ```suggestion
     - Select `patient-management-service` as the `Ballerina Project Directory`.
```

>      - Click Create. Once the component creation is complete, you will see the build page.

Reviewer: ```suggestion
     - Click **Create**. Once the component is created, the initial build will trigger automatically, generating a container image.
```

>      - Initial build will be triggered automatically and an image will be created.

Reviewer: We can remove this since the info is provided in the step above.

2. Build and deploy the component

>      - Once the build is completed, go to `Deploy` page. You can find it in the side drawer.

Reviewer: ```suggestion
     - After the build completes, go to the left navigation menu and click **Deploy**. 
```

     - Deploy the service.

## Test the service

> You can test the service in the `Test Console` Page. You can find it in the side drawer.

Reviewer: ```suggestion
You can go to the left navigation menu, click **Test**, and use the OpenAPI Console to test the service endpoints.
```
